Loggers fail to save to file if it already exists and it's the first time saving

The loggers will raise the following error when calling `saveLogFile()` if a log file with the same name exists and the logger has never been called before.

```java
Nov 05, 2017 10:34:37 AM ch.epfl.leb.sass.commandline.CommandLineInterfa
ce main                                                                
SEVERE: EvalError while executing shell script.
Sourced file: scripts/example_loggers.bsh : Method Invocation positionLo
gger.saveLogFile : at Line: 131 : in file: scripts/example_loggers.bsh :
 positionLogger .saveLogFile ( )                                       

Target exception: java.io.IOException: No such file or directory

        at bsh.BSHMethodInvocation.eval(BSHMethodInvocation.java:91)
        at bsh.BSHPrimaryExpression.eval(BSHPrimaryExpression.java:96)
        at bsh.BSHPrimaryExpression.eval(BSHPrimaryExpression.java:41)
        at bsh.Interpreter.eval(Interpreter.java:659)
        at bsh.Interpreter.source(Interpreter.java:577)
        at bsh.Interpreter.source(Interpreter.java:590)
        at ch.epfl.leb.sass.commandline.CommandLineInterface.main(Comman
dLineInterface.java:104)                                               
Caused by: java.io.IOException: No such file or directory
        at java.io.UnixFileSystem.createFileExclusively(Native Method)
        at java.io.File.createNewFile(File.java:1012)
        at ch.epfl.leb.sass.simulator.loggers.PositionLogger.saveLogFile
(PositionLogger.java:104)                                              
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAcces
sorImpl.java:62)                                                       
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMet
hodAccessorImpl.java:43)                                               
        at java.lang.reflect.Method.invoke(Method.java:498)
        at bsh.Reflect.invokeMethod(Reflect.java:131)
        at bsh.Reflect.invokeObjectMethod(Reflect.java:77)
        at bsh.Name.invokeMethod(Name.java:852)
        at bsh.BSHMethodInvocation.eval(BSHMethodInvocation.java:69)
        ... 6 more
```

The loggers will gracefully avoid an error by appending a number to the filename, but only if the the `saveLogFile()` method has already been called.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -30,6 +30,8 @@ Contains [ALICA v0.2.4]
 - The latency bug that caused a pause during every tenth simulation
   frame in GUI mode has been fixed. The latency came from updating the
   original status plots, which have been changed in this version.
+- Loggers now properly check for filename collisions to prevent
+  overwriting previously saved data.
 
 ## [v0.5.1]
 

--- a/src/ch/epfl/leb/sass/simulator/loggers/AbstractLogger.java
+++ b/src/ch/epfl/leb/sass/simulator/loggers/AbstractLogger.java
@@ -18,6 +18,8 @@
 package ch.epfl.leb.sass.simulator.loggers;
 
 import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.FileSystems;
 import java.io.IOException;
 
 /**
@@ -80,7 +82,13 @@ public abstract class AbstractLogger {
             if (2 == tokens.length)
                 baseName += "." + tokens[1];
             
-            baseName = logFile.getParent() + File.separator + baseName;
+            String parent = logFile.getParent();
+            if (parent == null) {
+                // Get current working directory in case no path was supplied
+                Path path = FileSystems.getDefault().getPath(".");
+                parent = path.toString();
+            }
+            baseName = parent + File.separator + baseName;
             logFile = new File(baseName);
             fileId++;
         }

